crates/sui-core/src/test_utils.rs

```diff
@@ -266,6 +266,28 @@ pub fn make_transfer_sui_transaction(
     to_sender_signed_transaction(data, keypair)
 }
 
+pub fn make_pay_sui_transaction(
+    gas_object: ObjectRef,
+    coins: Vec<ObjectRef>,
+    recipients: Vec<SuiAddress>,
+    amounts: Vec<u64>,
+    sender: SuiAddress,
+    keypair: &AccountKeyPair,
+    gas_price: Option<u64>,
+) -> VerifiedTransaction {
+    let data = TransactionData::new_pay(
```

Comment:
@tnowacki: One of the things I ran into here is that if `gas_object` is part of `coins`, validation will fail with `MutableObjectUsedMoreThanOnce`. This makes sense, but wondering if it's desirable for `new_pay` to prevent this by construction.